accounts-db/src/accounts_hash.rs

```diff
@@ -422,13 +335,21 @@ impl CumulativeHashesFromFiles {
         self.cumulative.total_count
     }
 
-    // return the biggest slice possible that starts at the overall index 'start'
-    fn get_slice(&self, start: usize) -> &[Hash] {
+    // return the biggest hash data possible that starts at the overall index 'start'
+    // start is the index of hashes
+    fn get_data(&self, start: usize) -> Arc<Vec<u8>> {
```

Comment:
can you help me understand the reason for returning `Vec<u8>` here and not `Vec<Hash>`. I know it will require some casting somewhere.

Reply:
The cast is happening inside the caller - `fn compute_merkle_root_from_start<F, T>`, where T is the type Hash. 

My thinking is that function is a low function to just return the raw bytes. And let the caller to cast it to the type its want. In a hypothetical case, the caller may want to use a different bytes for Hash. Then, it can pass a different T.

Reply:
pushed a commit to share the hash calcuation from slice and from owned vec.

Reply:
yes. done - change get_data() to return Box<[Hash]>. 9cc33668103c195d65f00986f3cd6ca52e8141ef